### PR TITLE
Sprint 8

### PR DESCRIPTION
**Pull request by leiasouza**
Testando os endpoints das APIs Account e Product.

### CODE REVIEW

File: account/test/e2e/createUserAccount.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { app } from "../../src/app.js";
+import request from "supertest";
+import { client, getUsersCollection } from "../../src/repositories/accountRepository.js";
+
+
+
+describe("POST em accounts", () => {
+  afterEach( async() => {
+    await client.connect();
```
Comment (DPechetti) [inline account/test/e2e/createUserAccount.test.js:9]: do you think this is the right place to create the connection?

File: product/src/test/e2e/createProduct.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { productExample } from "../data/products.js";
+import { app } from "../../app.js";
+import request from "supertest";
+import client from "../../repositories/databaseClients.js";
+
+describe("Criação de produto", () => {
+  afterAll(async() => {
+    await client.close();
```
Comment (DPechetti) [inline product/src/test/e2e/createProduct.test.js:8]: why are you closing the connection here?

File: account/src/repositories/accountRepository.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { MongoClient } from 'mongodb';
 
-const client = new MongoClient('mongodb://vanderleia:123456@account_db:27017');
+export const client = new MongoClient('mongodb://vanderleia:123456@localhost:27017');
```
[the remark this answers is not in the document]
Comment (leiasouza) [reply account/src/repositories/accountRepository.js:3]: I created an .env file to solve this.

File: account/test/e2e/createUserAccount.test.js (hunk 1 above)
Comment (leiasouza) [reply account/test/e2e/createUserAccount.test.js:9]: as a good practice, after each test always connect, clean and close the bank, so as not to interfere with other tests.

File: product/src/test/e2e/createProduct.test.js (hunk 2 above)
Comment (leiasouza) [reply product/src/test/e2e/createProduct.test.js:8]: here it was my lack of attention, the statement asked to clean the database and not close it, I already removed that line of code.

**Review by DPechetti — APPROVED** [review]

File: product/src/test/e2e/createProduct.test.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          createdAt: expect.any(String),
+          updatedAt: expect.any(String),
+          fetures: productExample.fetures.map((feture) => ({
+            ...feture,
+            id: expect.any(Number),
+            product_id: body.id,
+            createdAt: expect.any(String),
+            updatedAt: expect.any(String),
+          })),
+          images: productExample.images.map((image) => ({
```
Comment (wandersonmaceds) [inline product/src/test/e2e/createProduct.test.js:31]: passou sem o arrayContaining?

File: product/src/test/e2e/listProduct.test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import request from "supertest";
+
+describe("Lista produtos", () => {
+  it("Lista de produtos cadastrados", async () => {
+    await request(app)
+      .get("/products")
+      .then((response) => {
+        expect(response.status).toEqual(200);
+      });
+  });
```
Comment (wandersonmaceds) [inline product/src/test/e2e/listProduct.test.js:11]: faltou verificar se o corpo era um array vazio mesmo...

File: product/src/test/e2e/createProduct.test.js (hunk 4 above)
Comment (leiasouza) [reply product/src/test/e2e/createProduct.test.js:31]: Segue alterações com arrayContaining  na sprint 9.

File: product/src/test/e2e/listProduct.test.js (hunk 5 above)
Comment (leiasouza) [reply product/src/test/e2e/listProduct.test.js:11]: Abri uma issue para o cenário de testes de listar array vazio e cheio, subindo agora na sprint 9.

**Pull request merged**